core/src/main/java/com/linecorp/armeria/common/AggregatedHttpObject.java

```diff
@@ -36,9 +36,20 @@ public interface AggregatedHttpObject {
     HttpData content();
 
     /**
-     * Returns the content of this message as a string encoded in the specified {@link Charset}.
+     * Returns the content decoded in the {@link Charset} of the {@code content-type} header if present,
+     * or the content decoded in the specified {@link Charset} otherwise.
      */
     default String content(Charset charset) {
+        final MediaType contentType = headers().contentType();
+
+        if (contentType != null) {
+            final Charset contentTypeCharset = contentType.charset();
+
+            if (contentTypeCharset != null) {
+                return content(contentTypeCharset);
```

Comment:
Can you add a test which verifies this behavior?

Reply:
@jrhee17 I'm confused about the test class where the test code will be located.

Reply:
Should I make a new test class for AggregatedHttpObject Interface?

Reply:
You can create a new test class if necessary.

Reply:
@jrhee17 @ikhoon I've added test code.